keps/sig-windows/4885-windows-cpu-and-memory-affinity/README.md

```diff
@@ -0,0 +1,1006 @@
+
+| Kubelet Term | Description | Cadvisor term | Windows term |
+| --- | --- | --- | --- |
+| CPU | logical CPU | thread | Logical processor |
+| Core | physical CPU | Core | Core |
+| Socket | socket | Socket | Physical Processor |
+| NUMA Node | NUMA cell | Node | Numa node |
+
+The result of this mapping  gives the following output from CPU manager after the conversion into kubelet's memory structure:
+
+```json
+"Detected CPU topology" 
+topology={"NumCPUs":8,"NumCores":4,"NumSockets":1,"NumNUMANodes":1,"CPUDetails":{
+"0":{"NUMANodeID":0,"SocketID":1,"CoreID":0},
+"1":{"NUMANodeID":0,"SocketID":1,"CoreID":0},
+"2":{"NUMANodeID":0,"SocketID":1,"CoreID":2},
+"3":{"NUMANodeID":0,"SocketID":1,"CoreID":2},
+"4":{"NUMANodeID":0,"SocketID":1,"CoreID":4},
+"5":{"NUMANodeID":0,"SocketID":1,"CoreID":4},
+"6":{"NUMANodeID":0,"SocketID":1,"CoreID":6},
+"7":{"NUMANodeID":0,"SocketID":1,"CoreID":6}}}
+```
+
+The Windows API's used will be
+-	[getlogicalprocessorinformationex](https://learn.microsoft.com/windows/win32/api/sysinfoapi/nf-sysinfoapi-getlogicalprocessorinformationex)
+-	[nf-winbase-getnumaavailablememorynodeex](https://learn.microsoft.com/windows/win32/api/winbase/nf-winbase-getnumaavailablememorynodeex) 
+
+One difference between the Windows API and Linux is the concept of [Processor groups](https://learn.microsoft.com/windows/win32/procthread/processor-groups).
+On Windows systems with more than 64 cores the CPU's will be split into groups, 
+each processor is identified by its group number and its group-relative processor number. 
+
+In Cri we will add the following structure to the `WindowsContainerResources` in CRI:
+
+```protobuf
+message WindowsCpuGroupAffinity {
+    // CPU mask relative to this CPU group.
+    uint64 cpu_mask = 1;
+    // CPU group that this CPU belongs to.
+    uint32 cpu_group = 2;
+}
+```
+
+Since the Kubelet API's are looking for a distinct ProcessorId, the processorid's will be calculated by looping 
+through the mask and calculating the ids with `(group *64) + procesorid` resulting in unique processor id's from `group 0` as `0-63` and 
+processor Id's from `group 1` as `64-127` and so on. This translation will be done only in kubelet, the `cpu_mask` will be used when 
+communicating with the container runtime.
+
+```golang
+for i := 0; i < 64; i++ {
+		if GROUP_AFFINITY.Mask&(1<<i) != 0 {
+			processors = append(processors, i+(int(a.Group)*64))
+		}
+	}
+}
+```
+
+Using this logic, a cpu bit mask of `0000111` (leading zero's removed) would result in cpu's: 
+
+- `0,1,2` in `group 0` 
+- `64,65,66` in `group 1`.
+
+When converting back to the Windows Group Affinity we will divide the cpu number by 64 to get the group number then 
+use mod of 64 to calculate the location of the cpu in mask:
+
+```golang
+group := cpu / 64
+mask := 1 << (cpu % 64)
+groupaffinity.Mask |= mask
```

Comment:
Nit: One suggestion I have is to clarify variable names. You have GROUP_AFFINITY.Mask in one spot and groupaffinity.Mask here.